drf_spectacular/utils.py

```diff
@@ -173,15 +215,44 @@ def get_auth(self):
                     return auth
                 return super().get_auth()
 
+            def _get_examples(self, exclude_only):
+                if examples:
+                    return {
+                        title: example for title, example in examples.items()
+                        if getattr(example, exclude_only, None) is False
+                    }
+
             def get_request_serializer(self):
                 if request is not empty and is_in_scope(self):
+                    request_examples = self._get_examples(exclude_only='response_only')
+                    if request_examples:
+                        setattr(request, "SWAGGER_EXAMPLES", request_examples)
                     return request
-                return super().get_request_serializer()
+                serializer = super().get_request_serializer()
+
+                request_examples = self._get_examples(exclude_only='response_only')
+                if request_examples:
+                    setattr(serializer, "SWAGGER_EXAMPLES", request_examples)
+                return serializer
 
             def get_response_serializers(self):
                 if responses is not empty and is_in_scope(self):
+                    response_examples = self._get_examples(exclude_only='request_body_only')
```

Comment:
this whole section adds a lot of noise. pretty sure this can be made a bit more streamlined.